app/controllers/webview/drug_stocks_controller.rb

```diff
@@ -69,9 +77,13 @@ def safe_params
 
   def set_for_end_of_month
     @for_end_of_month ||= if params[:for_end_of_month]
-      Date.parse(params[:for_end_of_month])
+      Date.parse(params[:for_end_of_month]).end_of_month
```

Comment:
oof. good catch.

Reply:
ohey, didn't we have this method extracted to a module somewhere? precisely so we don't run into this kind of fragmentation issues?

Reply:
There were enough differences I was finding between the web view side and the admin side that extracting did not make sense (yet).